src/tool_operate.c

```diff
@@ -300,6 +300,8 @@ static CURLcode pre_transfer(struct GlobalConfig *global,
     }
     if(per->infd == -1)
 #else
+  if(per->uploadfile) {
+    if(!stdin_upload(per->uploadfile))
       per->infd = open(per->uploadfile, O_RDONLY | O_BINARY);
     if((per->infd == -1) || fstat(per->infd, &fileinfo))
```

Comment:
I need to refactor this a bit better, or this will close stdin if `fstat` fails here

Reply:
Also `per->infdopen` should be set to true only if uploadfile is not stdin